tests/python/frontend/onnx/test_forward.py

```diff
@@ -1404,191 +1424,181 @@ def verify_upsample3d_trilinear():
+            "Mean_test",
+            inputs=[
+                helper.make_tensor_value_info("a_np1", TensorProto.FLOAT, list(input_dim)),
+                helper.make_tensor_value_info("a_np2", TensorProto.FLOAT, list(input_dim)),
+                helper.make_tensor_value_info("a_np3", TensorProto.FLOAT, list(input_dim)),
+            ],
+            outputs=[helper.make_tensor_value_info("out", TensorProto.FLOAT, list(input_dim))],
+        )
 
+        model = helper.make_model(graph, producer_name="Mean_test")
+        verify_with_ort_with_inputs(model, [a_np1, a_np2, a_np3], target=target, dev=dev)
 
-@tvm.testing.uses_gpu
-def test_forward_mean():
     verify_mean((1, 3, 20, 20))
     verify_mean((20, 20))
 
 
-def verify_hardsigmoid(input_dim, alpha, beta):
-    dtype = "float32"
-
-    a_np1 = np.random.uniform(size=input_dim).astype(dtype)
+@tvm.testing.parametrize_targets
+def test_forward_hardsigmoid(target, dev):
+    def verify_hardsigmoid(input_dim, alpha, beta):
+        dtype = "float32"
 
-    hardsigmoid_node = helper.make_node("HardSigmoid", ["a_np1"], ["out"], alpha=alpha, beta=beta)
+        a_np1 = np.random.uniform(size=input_dim).astype(dtype)
 
-    graph = helper.make_graph(
-        [hardsigmoid_node],
-        "HardSigmoid_test",
-        inputs=[helper.make_tensor_value_info("a_np1", TensorProto.FLOAT, list(input_dim))],
-        outputs=[helper.make_tensor_value_info("out", TensorProto.FLOAT, list(input_dim))],
-    )
+        hardsigmoid_node = helper.make_node(
+            "HardSigmoid", ["a_np1"], ["out"], alpha=alpha, beta=beta
+        )
 
-    model = helper.make_model(graph, producer_name="HardSigmoid_test")
-    verify_with_ort_with_inputs(model, [a_np1])
+        graph = helper.make_graph(
+            [hardsigmoid_node],
+            "HardSigmoid_test",
+            inputs=[helper.make_tensor_value_info("a_np1", TensorProto.FLOAT, list(input_dim))],
+            outputs=[helper.make_tensor_value_info("out", TensorProto.FLOAT, list(input_dim))],
+        )
 
+        model = helper.make_model(graph, producer_name="HardSigmoid_test")
+        verify_with_ort_with_inputs(model, [a_np1], target=target, dev=dev)
 
-@tvm.testing.uses_gpu
-def test_forward_hardsigmoid():
     verify_hardsigmoid((1, 3, 20, 20), 0.5, 0.6)
     verify_hardsigmoid((20, 20), 0.3, 0.4)
 
 
-def verify_argreduce(input_dim, op_name, axis=None, keepdims=None):
-    a_np1 = np.random.uniform(-10, 10, input_dim).astype(np.int32)
-    out_shape = list(a_np1.shape)
-    def_axis = axis if axis is not None else 0
-    if keepdims == 1 or keepdims == None:
-        out_shape[def_axis] = 1
-    else:
-        out_shape.pop(def_axis)
-
-    node = onnx.helper.make_node(op_name, inputs=["a_np1"], outputs=["out"])
+# TODO (mbrookhart, electriclilies) Fix argmin on GPU and enable this test
+@tvm.testing.parametrize_targets
+def test_forward_arg_min_max(target, dev):
+    if "cuda" in target:
+        pytest.skip("Fails on CUDA")
```

Comment:
```suggestion
@tvm.testing.known_failing_targets("cuda")
@tvm.testing.parametrize_targets
def test_forward_arg_min_max(target, dev):
```

Perhaps consider using known_failing_targets here and elsewhere

Reply:
I tried that on multiple tests, and they still ran on cuda. This decorator apparently doesn't seem to work.

Reply:
Can you try using `@tvm.testing.exclude_targets('cuda')` instead?  The `known_failing_targets` applies `pytest.mark.xfail`, which still runs a test to see if it succeeds, but doesn't count a failure of that test as a failure of the test suite.  This makes sense if it is an assertion failure, but not if the test results in a segfault.  The `exclude_targets` decorator removes those tests entirely from the ones being run.

Reply:
`@tvm.testing.exclude_targets('cuda')` fails in the same way, test runs, fails, and is reported as a failure.

Reply:
Follow-up, it looks like currently the `@tvm.testing.parametrize_targets` decorator doesn't respect the settings in `known_failing_targets` or `exclude_targets`.  There's a slightly different code-path between the explicit parametrization and the auto-parametrization.  This is resolved with #8542, which merges these two paths.